go/service/rekey.go

```diff
@@ -111,7 +117,66 @@ func (h *RekeyHandler) DebugShowRekeyStatus(ctx context.Context, sessionID int)
 }
 
 func (h *RekeyHandler) RekeyStatusFinish(ctx context.Context, sessionID int) (keybase1.Outcome, error) {
-	return h.gregor.RekeyStatusFinish(ctx, sessionID)
+	outcome, err := h.gregor.RekeyStatusFinish(ctx, sessionID)
+	if err == nil {
+		// recheck rekey status 24h from now
+		h.recheckMu.Lock()
+		h.recheckDeadline = time.Now().Add(24 * time.Hour)
+		h.recheckMu.Unlock()
+	}
+	return outcome, err
+}
+
+func (h *RekeyHandler) recheckRekeyStatusPeriodic() {
+	h.G().Log.Debug("starting recheck rekey status loop")
+	ticker := time.NewTicker(1 * time.Hour)
+	h.G().PushShutdownHook(func() error {
+		h.G().Log.Debug("stopping recheckRekeyStatus timer")
+		ticker.Stop()
+		return nil
+	})
+	go func() {
+		for {
+			<-ticker.C
+			h.recheckMu.Lock()
+			if h.recheckDeadline.IsZero() {
+				h.recheckMu.Unlock()
```

Comment:
is there some way you can unlock only once?  i worry about having multiple unlocks in multiple branches, since it makes it harder to change the code in the future